application/src/main/java/org/opentripplanner/model/TripTimeOnDate.java

```diff
@@ -80,6 +81,34 @@ public static List<TripTimeOnDate> fromTripTimes(Timetable table, Trip trip) {
     return out;
   }
 
+  /**
+   * Must pass in both Timetable and Trip, because TripTimes do not have a reference to
+   * StopPatterns.
+   * <br>
+   * If the timetable does not contain the trip, scheduledTimetable is used instead.
+   *
+   * @param table the timetable for the service day
+   * @param serviceDate service day to set
+   */
+  public static List<TripTimeOnDate> fromTripTimesWithScheduleFallback(
+    Timetable table,
+    Trip trip,
+    LocalDate serviceDate,
+    Instant midnight,
+    TransitService transitService
+  ) {
```

Comment:
We could discuss in today's dev meeting if we want to move this method to DefaultTransitService to avoid having TransitService dependency here.